### ISSUE TITLE
Missing Sentinel

### ISSUE THREAD

**Issue by Electronicks** [JibbSmart/GamepadMotionHelpers#1]
Header files should either have sentinel define or _#pragma once_ to prevent multiple inclusion.

**Comment by JibbSmart** [comment]
Thanks! Will do

**Comment by Electronicks** [comment]
Actually, even with the sentinel, if the hpp file is included in multiple source files there will be linking issues because the same function is defined multiple times.

To my understanding, hpp is used for templated classes where implementation has to be part of the header, but GamepadMotion is not templated. So a regular pair .h/.cpp might be best.

**Comment by JibbSmart** [comment]
Won't #pragma once suffice? I'm keen to have it be only one file. I'm also going to look into transpiling soon (via Haxe) so it can just as easily be used in C#. I like the idea of having one file per language.

**Comment by Electronicks** [comment]
Pragma will prevent multiple declarations in a single source files at compile time,
But you can still run into multiple definitions in different sources at link time.
You could always templatize float away: someone may desire to use double for higher precision, or maybe some fixed point type.

**Comment by NullMember** [comment]
> Won't #pragma once suffice? I'm keen to have it be only one file. I'm also going to look into transpiling soon (via Haxe) so it can just as easily be used in C#. I like the idea of having one file per language.

I've tried to rewrite this project in Haxe. I've only wrote Quat and Vec classes and don't like result. If you still want to do something similiar, I can share it with you.
I think compiling this header into dynamic libraries like JoyShockLibrary is much more convenient for the job. I've done something similiar today (you can see it from my fork), but you can do it way cleaner than me.
Also, thanks for both GamepadMotionHelpers and JoyShockLibrary. Few years ago, I've tried to do something like them on my own but never achieved to success.

**Comment by kriNon** [comment]
Yeah, I can confirm that I ran into issues with multiple definitions in different sources at link time. Fixed it in my local copy by splitting into a cpp and h file.
